Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -191,24 +243,62 @@ internal class ResponseAccumulator: HTTPClientResponseDelegate {
     }
 }
 
-/// This delegate is strongly held by the HTTPTaskHandler
-/// for the duration of the HTTPRequest processing and will be
-/// released together with the HTTPTaskHandler when channel is closed
+/// HTTPClientResponseDelegate allows to receive notification about request processing and to control how reponse parts are processed.
+///
+///  - note: This delegate is strongly held by the `HTTPTaskHandler`
+///          for the duration of the `Request` processing and will be
+///          released together with the `HTTPTaskHandler` when channel is closed.
 public protocol HTTPClientResponseDelegate: AnyObject {
     associatedtype Response
 
+    /// Called when request `Head` is sent. Will be called once.
+    ///
+    /// - parameters:
+    ///     - task: Current request context.
+    ///     - head: Request `Head`.
     func didSendRequestHead(task: HTTPClient.Task<Response>, _ head: HTTPRequestHead)
 
+    /// Called when request part was sent. Could be called zero or more times.
+    ///
+    /// - parameters:
+    ///     - task: Current request context.
+    ///     - part: Request body `Part`.
     func didSendRequestPart(task: HTTPClient.Task<Response>, _ part: IOData)
 
+    /// Called when request is fully sent. Will be called once.
+    ///
+    /// - parameters:
+    ///     - task: Current request context.
     func didSendRequest(task: HTTPClient.Task<Response>)
 
+    /// Called when response `Head` is received. Will be called once. Response processing will resume when returned `EventLoopFuture` is fullfilled.
+    ///
+    /// - parameters:
+    ///     - task: Current request context.
+    ///     - head: Received reposonse head.
+    /// - returns: `EventLoopFuture` that will be used for backpressure.
     func didReceiveHead(task: HTTPClient.Task<Response>, _ head: HTTPResponseHead) -> EventLoopFuture<Void>
 
+    /// Called when response body `Part` is received. Could be called zero or more times. Response processing will resume when returned `EventLoopFuture` is fullfilled.
```

Comment:
"Called when part of a response body is received. Could be called zero or more times. You must return an `EventLoopFuture<Void>` that you complete when you have finished processing the body part. You can create an already succeeded future by calling `task.eventLoop.makeSucceededFuture(())`."

Reply:
fixed, thanks!